### PR TITLE
[Doc] Add *Manage Schema* Section

### PR DESCRIPTION
**Pull request by Anonymitaet**
Structure of Schema Chapter: https://github.com/apache/pulsar/issues/4789

### CODE REVIEW

File: site2/docs/schema-manage.md [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+  * If a **producer carries a schema**:
+  
+    A broker performs the compatibility check based on the configured compatibility check strategy of the namespace to which the topic belongs. 
+    
+    * If it is a new schema and it passes the compatibility check, the broker registers a new schema automatically for the topic.
+
+    * If the schema does not pass the compatibility check, the broker does not register a schema.
+
+![AutoUpdate Producer](assets/schema-autoupdate-producer.png)
+
```
Comment (Anonymitaet) [inline site2/docs/schema-manage.md]: @sijie Is this diagram accurate?
If so, I'll add one for `AutoUpdate for consumer`.

File: site2/docs/schema-manage.md [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+* If a **consumer connects to a topic with a schema**:
+
+  * If the **topic is idle** (no producers, no entries, no other consumers and no registered schemas), the broker registers the schema for the topic automatically.
+
+  * If the **topic is not idle**, the broker verifies if the schema provided by the consumer is compatible with the registered schema of the topic. 
+    
+    * If the **schema passes the compatibility check**, the consumer can connect to the topic and receive messages. 
+    
+    * If the **schema does not pass the compatibility check**, the consumer is rejected and disconnected.
```
Comment (Anonymitaet) [inline site2/docs/schema-manage.md]: @sijie Is this sentence accurate?

File: site2/docs/schema-manage.md [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+To use your custom schema storage implementation, you can perform the following steps.
+
+1. Package the implementation in a [JAR](https://docs.oracle.com/javase/tutorial/deployment/jar/basicsindex.html) file.
+   
+2. Add the JAR file to the `lib` folder in your Pulsar binary or source distribution.
+   
+3. Change the `schemaRegistryStorageClassName` configuration in `broker.conf` to your custom factory class.
+
+    For example, change the `schemaRegistryStorageClassName` to the `SchemaStorageFactory` implementation rather than the `SchemaStorage` implementation.
+
```
Comment (Anonymitaet) [inline site2/docs/schema-manage.md]: @sijie Is the sentence accurate?

File: site2/docs/schema-manage.md [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+    
+    * If `isSchemaValidationEnforced` or `schemaValidationEnforced` is **enabled** in the namespace to which the topic belongs, the producer is rejected and disconnected.
+
+  * If a **producer carries a schema**:
+  
+    A broker performs the compatibility check based on the configured compatibility check strategy of the namespace to which the topic belongs. 
+    
+    * If it is a new schema and it passes the compatibility check, the broker registers a new schema automatically for the topic.
+
+    * If the schema does not pass the compatibility check, the broker does not register a schema.
```
Comment (Anonymitaet) [inline site2/docs/schema-manage.md]: @sijie Is line 31 accurate?

**Comment by Anonymitaet** [comment]
@sijie 

I've tested and found the `extract` command is not documented yet, should I add this?
If so, could you please provide some descriptions (I do not know what it is)?

WX20190730-232949@2x

File: site2/docs/schema-manage.md (hunk 4 above)
Comment (sijie) [reply site2/docs/schema-manage.md]: that's correct.

File: site2/docs/schema-manage.md [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+To manage schemas, you can use one of the following methods:
+
+* **Admin CLI** (`pulsar-admin` CLI)
+  
+* **REST API** (Pulsar exposes schema related management API in Pulsar’s admin RESTful API. You can access the admin RESTful endpoint directly to manage schemas)
+
+* **Java Admin API** (use Java admin library provided by Pulsar)
+
+### Upload a schema
```
Comment (sijie) [inline site2/docs/schema-manage.md]: Can we organize the `admin-cli`, `restful-api` and `java-admin` in the multi-tab way?

for example: http://pulsar.apache.org/docs/en/next/functions-develop/

File: site2/docs/schema-manage.md (hunk 3 above)
Comment (sijie) [reply site2/docs/schema-manage.md]: I am not sure that is correct. I think we can remove it.

Comment (Anonymitaet) [reply site2/docs/schema-manage.md]: OK

File: site2/docs/schema-manage.md (hunk 5 above)
Comment (Anonymitaet) [reply site2/docs/schema-manage.md]: Thanks for your reminding.
Done

**Comment by Anonymitaet** [comment]
@sijie Thanks for your comments, I've incorporated all of them.
Could you PTAL?

**Comment by Anonymitaet** [comment]
run java8 tests

**Pull request merged**